### PR TITLE
Highlight Admission controller+prep 1.20

### PR DESCRIPTION
**Pull request by yoannrobin**
### What does this PR do?

* Highlight Admission controller in APM kubernetes setup
* Update Admission controller doc with new configuration modes (Cluster Agent 1.20)

### Preview

https://docs-staging.datadoghq.com/yrobin/UDS_APM_enabled/agent/cluster_agent/admission_controller/
https://docs-staging.datadoghq.com/yrobin/UDS_APM_enabled/agent/kubernetes/apm/?tab=helm

### Additional Notes

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

File: content/en/agent/cluster_agent/admission_controller.md [hunk 1]
```diff
@@ -145,10 +145,32 @@ Possible options:
...
+#### Configure APM and DogstatsD communication mode
+Starting from Datadog Cluster Agent v1.20.0, the Datadog Admission Controller can be configured to inject different mode of communication between the application and Datadog agent.
+
+This feature can be configured by setting `admission_controller.inject_config.mode` or by defining a pod-specific mode using the `admission.datadoghq.com/config.mode` pod label.
+
+Possible options:
+| Mode               | Description                                                                     |
+|--------------------|---------------------------------------------------------------------------------|
+| `hostip` (Default) | Inject the host IP in `DD_AGENT_HOST` environment variable                      |
+| `service`          | Inject Datadog's local-service DNS name in `DD_AGENT_HOST` environment variable |
```
[the remark this answers is not in the document]
Comment (clamoriniere) [reply content/en/agent/cluster_agent/admission_controller.md:166]: we have a documentation card on our side to document the `network local service` and the new admission controller capability when Agent 7.36 and DCA 1.20 are released.

[the remark this answers is not in the document]
Comment (hkaj) [reply content/en/agent/cluster_agent/admission_controller.md:166]: ok great thank you

File: content/en/agent/cluster_agent/admission_controller.md [hunk 2]
```diff
@@ -145,10 +145,31 @@ Possible options:
...
+#### Configure APM and DogstatsD communication mode
+Starting from Datadog Cluster Agent v1.20.0, the Datadog Admission Controller can be configured to inject different mode of communication between the application and Datadog agent.
+
+This feature can be configured by setting `admission_controller.inject_config.mode` or by defining a pod-specific mode using the `admission.datadoghq.com/config.mode` pod label.
+
+Possible options:
+| Mode               | Description                                                                                                       |
+|--------------------|-------------------------------------------------------------------------------------------------------------------|
+| `hostip` (Default) | Inject the host IP in `DD_AGENT_HOST` environment variable                                                        |
+| `service`          | Inject Datadog's local-service DNS name in `DD_AGENT_HOST` environment variable                                   |
```
Comment (ahmed-mez) [inline content/en/agent/cluster_agent/admission_controller.md:165]: perhaps we can add a note that this requires k8s v1.22+

File: content/en/agent/cluster_agent/admission_controller.md [hunk 3]
```diff
@@ -11,9 +11,9 @@ further_reading:
 ---
 
 ## Overview
-The Datadog admission controller is a component of the Datadog Cluster Agent. The main benefit of the admission controller is to simplify the user's application pod configuration. For that, It has two main functionalities:
+The Datadog admission controller is a component of the Datadog Cluster Agent. The main benefit of the admission controller is to simplify the user's application pod configuration. For that, it has two main functionalities:
```
Comment (jtappa) [inline content/en/agent/cluster_agent/admission_controller.md:14]:
```suggestion
The Datadog admission controller is a component of the Datadog Cluster Agent. The main benefit of the admission controller is to simplify your application pod configuration. For that, it has two main functionalities:
```

File: content/en/agent/cluster_agent/admission_controller.md [hunk 4]
```diff
@@ -145,10 +145,31 @@ Possible options:
 | `false`          | `admission.datadoghq.com/enabled=true`  | Yes       |
 | `false`          | `admission.datadoghq.com/enabled=false` | No        |
 
+
+#### Order of priority
+The Datadog admission controller does not inject the environment variables `DD_VERSION`, `DD_ENV`, and `DD_SERVICE` if they already exist.
+
+When these environment variables are not set, the admission controller will use standard tags value in the following order (highest first):
+- Labels on the pod
+- Labels on the ownerReference (ReplicaSets, DaemonSets, Deployments...)
```
Comment (jtappa) [inline content/en/agent/cluster_agent/admission_controller.md:154]:
```suggestion
When these environment variables are not set, the admission controller uses standard tags value in the following order (highest first):

- Labels on the pod
- Labels on the `ownerReference` (ReplicaSets, DaemonSets, Deployments...)
```

File: content/en/agent/cluster_agent/admission_controller.md [hunk 5]
```diff
@@ -145,10 +145,31 @@ Possible options:
...
+
+#### Configure APM and DogstatsD communication mode
+Starting from Datadog Cluster Agent v1.20.0, the Datadog Admission Controller can be configured to inject different mode of communication between the application and Datadog agent.
```
Comment (jtappa) [inline content/en/agent/cluster_agent/admission_controller.md:157]:
```suggestion
Starting from Datadog Cluster Agent v1.20.0, the Datadog Admission Controller can be configured to inject different modes of communication between the application and Datadog agent.
```

File: content/en/agent/kubernetes/apm.md [hunk 6]
```diff
@@ -163,6 +164,15 @@ $ kubectl apply -n $DD_NAMESPACE -f datadog-agent.yaml
 ### Configure your application pods to submit traces to Datadog Agent
 
 {{< tabs >}}
+
+{{% tab "Datadog Admission Controller" %}}
+[Datadog Admission Controller][1] is able to inject environment variables and mount the necessary volumes on new application-pods in order to configure pod/agent trace communication automatically.
```
Comment (jtappa) [inline content/en/agent/kubernetes/apm.md:169]:
```suggestion
[Datadog Admission Controller][1] is able to inject environment variables and mount the necessary volumes on new application-pods in order to configure pod and agent trace communication automatically.
```

File: content/en/agent/kubernetes/apm.md [hunk 7]
```diff
@@ -163,6 +164,15 @@ $ kubectl apply -n $DD_NAMESPACE -f datadog-agent.yaml
...
+
+To learn more on how to automatically configure you application to submit traces to Datadog Agent refer to [Datadog Admission Controller][1].
```
Comment (jtappa) [inline content/en/agent/kubernetes/apm.md:171]:
```suggestion
To learn more on how to automatically configure you application to submit traces to Datadog Agent read the [Datadog Admission Controller][1] documentation.
```

**Review by jtappa — CHANGES_REQUESTED** [review]

**Review by jtappa — APPROVED** [review]

**Pull request merged**